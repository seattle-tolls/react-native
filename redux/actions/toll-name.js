import { SWITCH_TOLL_NAME } from '../constants/action-types'

export const switchTollName = (payload) => ({
  type: SWITCH_TOLL_NAME, payload,
})