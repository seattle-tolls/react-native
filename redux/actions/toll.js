import { SET_TOLL_INFO, FETCH_TOLL_INFO } from '../constants/action-types'

export const setTollInfo = (payload) => ({
  type: SET_TOLL_INFO, payload,
})

export const fetchTollInfo = () => ({
  type: FETCH_TOLL_INFO,
})