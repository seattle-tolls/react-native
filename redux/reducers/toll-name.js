import { SWITCH_TOLL_NAME } from '../constants/action-types'
// import { createSelector } from 'reselect'

const initialState = ''

const tollNameReducer = (tollName = initialState, action) => {
  switch (action.type) {
    case SWITCH_TOLL_NAME:
      return action.payload
    default:
      return tollName
  }
}

// TODO: finish the get name selector
export const getName = state => {
  console.log('TOLLNAME-->', state.tollName)
}

export default tollNameReducer