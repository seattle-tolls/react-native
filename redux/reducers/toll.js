import { SET_TOLL_INFO } from '../constants/action-types'
import { createSelector } from 'reselect'

const initialState = {}

const tollReducer = (toll = initialState, action) => {
  switch (action.type) {
    case SET_TOLL_INFO:
      return action.payload
    default:
      return toll
  }
}

export const tollInfo = (state) => {
  const tollInfoObj = state.toll[state.tollName]
  console.log('TOLL INFO', tollInfoObj && tollInfoObj.name)
  return tollInfoObj
}

// FIXME: fix the other selectors according with the derived data

// export const todayInfo = state => state.toll.todayInfo
// export const todaySchedule = state => state.toll.todaySchedule

// export const todayScheduleAsArray = createSelector(
//   todaySchedule,
//   (schedule) => {
//     if(schedule[0]){
//       const scheduleArray = Object.keys(schedule).map(hour => {
//         return ({
//           id: hour,
//           goodToGo: schedule[hour].goodToGo,
//           payByMail: schedule[hour].payByMail,
//           schedule: schedule[hour].schedule,
//         })
//       })
//       return scheduleArray
//     }
//   }
// )

// export const currentTimeInfo = createSelector(
//   todayInfo,
//   todaySchedule,
//   (info, schedule) => {
//     for(let i = info.hour; i >= 0; i--){
//       if(schedule[i]){
//         return{
//           dayName: info.dayName,
//           holidayName: info.holidayName,
//           ...schedule[i],
//         }
//       }
//     }
//   }
// )

export default tollReducer
