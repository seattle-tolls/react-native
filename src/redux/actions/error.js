import { REPORT_ERROR } from '../constants/action-types'

export const reportError = (error) => ({
  type: REPORT_ERROR, error,
})