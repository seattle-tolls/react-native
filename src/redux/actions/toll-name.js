import { SET_TOLL_NAME } from '../constants/action-types'

export const setTollName = (payload) => ({
  type: SET_TOLL_NAME, payload,
})
