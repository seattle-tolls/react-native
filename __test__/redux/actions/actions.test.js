import { fetchTollInfo, setTollInfo } from '../../../src/redux/actions/toll'
import { setTollName } from '../../../src/redux/actions/toll-name'

describe('REDUX ACTIONS', () => {
  describe('Toll action', () => {
    it('should run fetch toll action', () => {
      const fetchToll = fetchTollInfo()
      expect(fetchToll).toMatchSnapshot()
    })

    it('should run set toll action', () => {
      const payload = { toll1: 'one', toll2: 'two' }
      const setToll = setTollInfo(payload)
      expect(setToll).toMatchSnapshot()
    })
  })

  describe('Toll Name action', () => {
    it('should run the toll name action', () => {
      const payload = 'fakeToll'
      const tollName = setTollName(payload)
      expect(tollName).toMatchSnapshot()
    })
  })
})