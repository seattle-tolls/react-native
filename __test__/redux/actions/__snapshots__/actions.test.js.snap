// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REDUX ACTIONS Toll Name action should run the toll name action 1`] = `
Object {
  "payload": "fakeToll",
  "type": "SET_TOLL_NAME",
}
`;

exports[`REDUX ACTIONS Toll action should run fetch toll action 1`] = `
Object {
  "type": "FETCH_TOLL_INFO",
}
`;

exports[`REDUX ACTIONS Toll action should run set toll action 1`] = `
Object {
  "payload": Object {
    "toll1": "one",
    "toll2": "two",
  },
  "type": "SET_TOLL_INFO",
}
`;
